tests/test_convert.py

```diff
@@ -0,0 +1,186 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import pathlib
+
+import flask
+import pytest
+
+from cloudevents.http import from_json
+
+from functions_framework import convert
+from functions_framework.background import Event
+from functions_framework.exceptions import EventConversionException
+from google.cloud.functions.context import Context
+
+TEST_DATA_DIR = pathlib.Path(__file__).resolve().parent / "test_data"
+
+
+PUBSUB_BACKGROUND_EVENT = {
+    "context": {
+        "eventId": "1215011316659232",
+        "timestamp": "2020-05-18T12:13:19Z",
+        "eventType": "google.pubsub.topic.publish",
+        "resource": {
+            "service": "pubsub.googleapis.com",
+            "name": "projects/sample-project/topics/gcf-test",
+            "type": "type.googleapis.com/google.pubsub.v1.PubsubMessage",
+        },
+    },
+    "data": {
+        "data": "10",
+    },
+}
+
+PUBSUB_BACKGROUND_EVENT_WITHOUT_CONTEXT = {
+    "eventId": "1215011316659232",
+    "timestamp": "2020-05-18T12:13:19Z",
+    "eventType": "providers/cloud.pubsub/eventTypes/topic.publish",
+    "resource": "projects/sample-project/topics/gcf-test",
+    "data": {
+        "data": "10",
+    },
+}
+
+
+@pytest.fixture
+def pubsub_cloudevent_output():
+    event = {
+        "specversion": "1.0",
+        "id": "1215011316659232",
+        "source": "//pubsub.googleapis.com/projects/sample-project/topics/gcf-test",
+        "time": "2020-05-18T12:13:19Z",
+        "type": "google.cloud.pubsub.topic.v1.messagePublished",
+        "datacontenttype": "application/json",
+        "data": {
+            "message": {
+                "data": "10",
+            },
+        },
+    }
+
+    return from_json(json.dumps(event))
+
+
+@pytest.fixture
+def firebase_auth_background_input():
+    with open(TEST_DATA_DIR / "firebase-auth-legacy-input.json", "r") as f:
+        return json.load(f)
+
+
+@pytest.fixture
+def firebase_auth_cloudevent_output():
+    with open(TEST_DATA_DIR / "firebase-auth-cloudevent-output.json", "r") as f:
+        return from_json(f.read())
+
+
+@pytest.mark.parametrize(
+    "event", [PUBSUB_BACKGROUND_EVENT, PUBSUB_BACKGROUND_EVENT_WITHOUT_CONTEXT]
+)
+def test_pubsub_event_to_cloudevent(event, pubsub_cloudevent_output):
+    req = flask.Request.from_values(json=event)
+    cloudevent = convert.background_event_to_cloudevent(req)
+    assert cloudevent == pubsub_cloudevent_output
+
+
+def test_firebase_auth_event_to_cloudevent(
+    firebase_auth_background_input, firebase_auth_cloudevent_output
+):
+    req = flask.Request.from_values(json=firebase_auth_background_input)
+    cloudevent = convert.background_event_to_cloudevent(req)
+    assert cloudevent == firebase_auth_cloudevent_output
+
+
+def test_split_resource():
```

Comment:
Would it make sense to parametrize some of these? Or would it be confusing to read?

Reply:
Probably so. I've parametrized three of them. I think the others would be harder to read.